options.go

```diff
@@ -338,7 +338,7 @@ func (o *Options) Validate() error {
 	msgs = parseProviderInfo(o, msgs)
 
 	var cipher *encryption.Cipher
-	if o.PassAccessToken || o.SetAuthorization || o.PassAuthorization || (o.CookieRefresh != time.Duration(0)) {
+	if o.CookieSecret != "" {
```

Comment:
We will need to have a check that ensures that if any of the options that were triggering this are set, that the cookie secret must not be empty, otherwise we could end up with unencrypted sessions containing sensitive information

Reply:
Running the proxy without a cookie secret was not possible to begin with. See a bit earlier `verify`: https://github.com/oauth2-proxy/oauth2-proxy/pull/414/files#diff-4aae454123f50b906f9aaef44f10011eR226-R228.

```
	if o.CookieSecret == "" {
		msgs = append(msgs, "missing setting: cookie-secret")
	}
```

`Verify()` could use some refactoring love to make this a bit more obvious.

Reply:
Given the fact we know the cookie-secret is set at this point, you could take this whole setup out of this if statement, it provides no extra value right? Just maybe add a comment to say that we've already verified the cookie-secret is set?